crates/distribution-types/src/annotation.rs

```diff
@@ -0,0 +1,52 @@
+use serde::{Deserialize, Deserializer, Serialize};
+use std::path::PathBuf;
+use uv_fs::Simplified;
+
+/// Source of a dependency, e.g., a `-r requirements.txt` file.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize)]
+pub enum SourceAnnotation {
+    /// A `-r requirements.txt` file.
+    Requirement(PathBuf),
+    /// A `-c constraints.txt` file.
+    Constraint(PathBuf),
+    /// An `--override overrides.txt` file.
+    Override(PathBuf),
```

Comment:
I made these all `PathBuf`, so that we convert at the very end (when we display) rather than eagerly.

Reply:
Ok. I'm fine with that, but I'm not sure how needed that is, as AFAIK at this point their only use is as something we'll print, so they could be eagerly converted. This works though.